packages/design-system/src/components/Legend/index.tsx

```diff
@@ -0,0 +1,59 @@
+import classNames from "classnames";
+import type { FC } from "react";
+import { GenericKeys } from "@helpers/types";
+import { LegendEntryProps, LegendProps } from "./Legend.interface";
+
+export const Legend: FC<LegendProps> = ({
+  entries,
+  horizontal = true,
+  background = "transparent",
+  ...props
+}) => {
+  return (
+    <ul
+      className={classNames(
+        "flex flex-wrap gap-4 p-2",
+        background,
+        !horizontal && "flex-col"
+      )}
+    >
+      {entries.map((entry, i) => (
+        <LegendEntry {...props} entry={entry} key={`${i}-${entry.label}`} />
+      ))}
+    </ul>
+  );
+};
+
+const LegendEntry: FC<LegendEntryProps> = ({
+  shape = "square",
+  displayValues,
+  entry,
+}) => {
+  const { label, color, value } = entry;
+
+  const labelShapeVariants: GenericKeys = {
+    circle: "rounded-full",
+    rounded: "rounded-sm",
+    square: "",
+  };
+
+  return (
+    <li
+      className={classNames(
+        "flex items-center gap-x-2 text-xs text-neutral-500"
+      )}
+    >
+      <div
+        className={classNames(
+          "aspect-square w-4",
+          color,
+          labelShapeVariants[shape]
+        )}
+      />
+      <div>
+        <span>{label}</span>
+        {displayValues && <span className="pl-1">{`(${value})`}</span>}
```

Comment:
let's get rid of this boolean. We can simply do sth like:
```suggestion
        <span className="pl-1 empty:hidden" >{`(${value})`}</span>
```